### ISSUE TITLE
Add `CallAllContracts` option to only call contracts specified in `DeploymentOrder`

### ISSUE THREAD

**Issue by ggrieco-tob** [crytic/medusa#132]
This example should NOT be solved if `allContracts` is false, since medusa should be allowed to call `C` (and not `A`). 

```solidity
interface iA {
  function produceFail() external;
}

contract A {
  function fail(address a) public {
    iA(a).produceFail();
  }
}


contract C {
  bool state = true;
  A a = new A();

  function pass() public {  assert(state); }
  function produceFail() public {
     require(msg.sender == address(a)); 
     state = false;
  } 
} 
```

However, using the command line:

```
$ medusa fuzz --target simple.sol --deployment-order C --assertion-mode
[FAILED] Assertion Test: C.pass()
Test for method "C.pass()" failed after the following call sequence resulted in an assertion:
1) A.fail(0xa647ff3c36cfab592509e13860ab8c4f28781a66) (block=3065, time=360624, gas=12500000, gasprice=1, value=0, sender=0x0000000000000000000000000000000000030000)
2) C.pass() (block=40515, time=721248, gas=12500000, gasprice=1, value=0, sender=0x0000000000000000000000000000000000030000)
 -> [Execution Trace]
 -> [call] C.pass() (addr=0xA647ff3c36cFab592509E13860ab8c4F28781a66, value=0, sender=0x0000000000000000000000000000000000030000)
	 -> [assertion failed]
```

Even using a medusa.json file, with `allContracts` set to false will not work.

**Comment by anishnaik** [comment]
I actually don't think `TestAllContracts` is implemented yet - I can take a deeper look tomorrow.

**Comment by Xenomega** [comment]
` TestAllContracts` is implemented correctly here I believe. We only run tests in `C`, but the way we chose to implement this, `medusa` will call _any_ contract it knows of existing to try and violate the property. This way medusa can find all the ways a test can be violated even indirectly.

I think maybe this issue should be translated into one like:

"Implement a `CallAllContracts` feature similar to `TestAllContracts` that allows for disabling of calls to non-deployment order contracts."

**Comment by ggrieco-tob** [comment]
> "Implement a CallAllContracts feature similar to TestAllContracts that allows for disabling of calls to non-deployment order contracts." so you can test using a different flow if you wanted to.

This is correct.

**Comment by ggrieco-tob** [comment]
I will start working on this to assign a new config option that replicates echidna behavior. After that, we can decide what to do exactly.

**Comment by ggrieco-tob** [comment]
A draft to solve this issue is available in #169

**Comment by anishnaik** [comment]
@ggrieco-tob agree that this is definitely needed. What we are going to do is merge this capability into `testAllContracts`. So, similar to `multiAbi`, if `testAllContracts` is false, all dynamically deployed contracts and stuff not in the deployment order will not be called. 

Closing this issue, since I will be creating a new one to specify the change to the config options.
